src/test/java/frc/robot/subsystems/swerve/PhoenixOdometryThreadTest.java

```diff
@@ -0,0 +1,91 @@
+package frc.robot.subsystems.swerve;
+
+import com.ctre.phoenix6.BaseStatusSignal;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Queues;
+import frc.robot.subsystems.swerve.PhoenixOdometryThread.Samples;
+import java.util.Collections;
+import java.util.Map;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class PhoenixOdometryThreadTest {
```

Comment:
not sure if we have time for it at this point in the season, but id love to go over unit testing and how to apply it to robot code at some point